### PR TITLE
Add the HDF5 discussion

### PR DESCRIPTION
**Pull request by dangars**
Hi, I added my name to the end of the list. Do you have a convention for the order in the author list?

I added as well the discussion about the HDF5 interface.

### CODE REVIEW

**Comment by Rombur** [comment]
> Do you have a convention for the order in the author list?

Yes, we use the alphabetical order.

File: 9.1/paper.tex [hunk 1]
```diff
@@ -173,7 +179,7 @@ \section{Overview}
     Section~\ref{subsec:sd}),
   \item Full support for $hp$~adaptivity in parallel computations (see
     Section~\ref{subsec:hp}),
-  \item Interfaces to the HDF5 file format and libraries (see
+  \item Interface to the HDF5 file format and libraries (see
```
Comment (bangerth) [inline 9.1/paper.tex]:
```suggestion
  \item An interface to the HDF5 file format and libraries (see
```

File: 9.1/paper.tex [hunk 2]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
 
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\subsection{Interfaces to the HDF5 file format and libraries}
+\subsection{Interface to the HDF5 file format and libraries}
 \label{subsec:hdf5}
-
-\marginpar{Missing}
+The new deal.II's \texttt{HDF5} interface allows to write \texttt{HDF5} files and manipulate
```
Comment (bangerth) [inline 9.1/paper.tex]: please use `\dealii{}`

File: 9.1/paper.tex [hunk 3]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
...
+datasets, groups and attributes in serial and  in parallel using MPI. The
```
Comment (bangerth) [inline 9.1/paper.tex]:
```suggestion
datasets, groups, and attributes in serial and  in parallel using MPI. The
```

File: 9.1/paper.tex [hunk 4]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
 
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\subsection{Interfaces to the HDF5 file format and libraries}
+\subsection{Interface to the HDF5 file format and libraries}
```
Comment (bangerth) [inline 9.1/paper.tex]: Can you also add an introductory sentence that explains what HDF5 is?

File: 9.1/paper.tex [hunk 5]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
...
 \label{subsec:hdf5}
-
-\marginpar{Missing}
+The new deal.II's \texttt{HDF5} interface allows to write \texttt{HDF5} files and manipulate
+datasets, groups and attributes in serial and  in parallel using MPI. The
+\texttt{HDF5} MPI library calls that modify the structure of the file are
+always collective, whereas writing and reading raw data in a dataset can be done
+independently or collectively. In the deal.II's \texttt{HDF5} interface all the calls are
```
Comment (bangerth) [inline 9.1/paper.tex]: same here and below

File: 9.1/paper.tex [hunk 6]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
...
+set to collective in order to maximize the performance. This means that all the MPI
```
Comment (bangerth) [inline 9.1/paper.tex]:
```suggestion
set to collective in order to maximize performance. This means that all the MPI
```

File: 9.1/paper.tex [hunk 7]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
...
+processes have to contribute to every single call, even if they don't have data to write.
+We have added the following classes to the deal.II's \texttt{HDF5} namespace. 
+\begin{itemize}
+\item The new \texttt{HDF5::File} class can be used to open and create \texttt{HDF5} files
+in serial or in parallel. 
+\item The new \texttt{HDF5::Group} class can be used to open and create groups.
+The \texttt{HDF5} files have a tree structure. The root contains groups and
```
Comment (bangerth) [inline 9.1/paper.tex]:
```suggestion
 \texttt{HDF5} files have a tree structure. The root contains groups and
```

File: 9.1/paper.tex [hunk 8]
```diff
@@ -478,10 +484,34 @@ \subsection{Full support for $hp$~adaptivity in parallel computations}
...
+the groups can contain other groups.
+\item The new \texttt{HDF5::DataSet} class can be used to open and create datasets which
+can be placed inside groups or at the root of the \texttt{HDF5} file. A dataset can be a
+vector a matrix or a tensor. It is possible to read and write in the dataset using
```
Comment (bangerth) [inline 9.1/paper.tex]:
```suggestion
vector, a matrix, or a tensor. It is possible to read and write in the dataset using
```

File: 9.1/paper.tex (hunk 4 above)
Comment (dangars) [reply 9.1/paper.tex]: I added an introductory paragraph of HDF5, is that ok?

**Comment by bangerth** [comment]
Thanks. By the way, if you have edits on the rest of the paper, feel free to propose them through a pull request -- you're just as much a co-author of the paper as anyone else! (The same applied to @GrahamBenHarper !)

**Comment by dangars** [comment]
@bangerth It looks like today I won't have time to review the paper, but tomorrow I'll give another review to the paper :)

**Pull request merged**